src/async_handler.h

```diff
@@ -48,7 +49,7 @@ namespace AsyncHandler {
 	 * @param file_name Name of the requested file requested.
 	 * @return The async request.
 	 */
-	FileRequestAsync* RequestFile(const std::string& folder_name, const std::string& file_name);
+	FileRequestAsync* RequestFile(StringView folder_name, StringView file_name);
```

Comment:
I wish I could tell you yes, replace all `const std::string&` with `StringView`. Unfortunately this is C++ so there is always one big caveat. 

`StringView` is not null terminated, so if at the end of your call stack you end up calling a C library which requires null terminated `const char*`, you'll end up having to make a copy so you can get a new string with a null byte at the end.

Aside from that caveat, yes use `StringView` where-ever you need a non-owning reference. Not only is it a generic adapter but the calling convention is faster. 

`const string&` if you think about it in C terms, is actually a double pointer `const char**`. The reference is a pointer to the string object, which contains another pointer to the bytes.

When you pass a `StringView` you're passing a `const char*`, removing one level of indirection. In some cases this can help the optimizer with it's aliasing analysis to produce faster code.

The same rule applies to `vector` and `Span`.